From 0af1270e7de1377a8476058a23abab8930b305cf Mon Sep 17 00:00:00 2001
From: Robert Yang <liezhi.yang@windriver.com>
Date: Sun, 18 May 2014 16:15:29 +0800
Subject: [PATCH 1/2] formats/fs.py: enable EFIFS in create fs box

The user should be able to creat a EFIFS partition for EFI boot, the
supported() function will check _labelfs,  _resizefs and _infofs, we can
use "mkdosfs -n" as the _labelfs, for _resizefs and _infofs, the EFIFS
doesn't support resize, so we just set them to the command "true" as a
workaround.

And set vfat to fat32 since we have used "-F 32" as
_defaultFormatOptions

Upstream-Status: Pending

Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
---
 pyanaconda/storage/formats/fs.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/pyanaconda/storage/formats/fs.py b/pyanaconda/storage/formats/fs.py
index bcf500d..dc2edc7 100644
--- a/pyanaconda/storage/formats/fs.py
+++ b/pyanaconda/storage/formats/fs.py
@@ -1098,7 +1098,7 @@ class FATFS(FS):
     _defaultFormatOptions = ["-F", "32"]
     _defaultMountOptions = ["umask=0077", "shortname=winnt"]
     # FIXME this should be fat32 in some cases
-    partedSystem = fileSystemType["fat16"]
+    partedSystem = fileSystemType["fat32"]
 
     def _fsckFailed(self, rc):
         if rc >= 1:
@@ -1118,6 +1118,11 @@ class EFIFS(FATFS):
     _name = "EFI System Partition"
     _minSize = 50
     _bootable = True
+    _labelfs = "mkdosfs"
+    _resizefs = "true"
+    _infofs = "true"
+    _resizable = False
+    _defaultLabelOptions = ["-n"]
 
     @property
     def supported(self):
-- 
1.8.3.1

